rmf_schedule_visualizer/src/VisualizerData.cpp

```diff
@@ -102,6 +102,32 @@ void VisualizerDataNode::start(Data _data)
       debug_cb(std::move(msg));
     });
 
+  // retrieve/construct mirrors, snapshots and negotiation object
+  {
+    _writer = rmf_traffic_ros2::schedule::Writer::make(*this);
+
+    using namespace std::chrono_literals;
+
+    bool ready = false;
+    const auto stop_time = std::chrono::steady_clock::now() + 10s;
+    while (rclcpp::ok() && std::chrono::steady_clock::now() < stop_time)
```

Comment:
When `_writer` is removed, this whole `while` loop can be reduced to simply

```
_negotiation = rmf_traffic_ros2::schedule::Negotiation(*this, data->mirror.snapshot_handle());
```

Everything else about the while loop can be thrown away.

Reply:
removed 3b1c580